packages/devtools_app/lib/src/memory/memory_treemap.dart

```diff
@@ -0,0 +1,334 @@
+typedef SelectNodeFunction = void Function(TreemapNode);
+
+class InstructionsSize {
+  const InstructionsSize(this.root);
+
+  factory InstructionsSize.fromSnapshop(MemoryController controller) {
+    final Map<String, TreemapNode> rootChildren = <String, TreemapNode>{};
+    final TreemapNode root = TreemapNode(
+      name: 'root',
+      childrenMap: rootChildren,
+    );
+    TreemapNode currentParent = root;
+
+    // TODO(terry): Should treemap be all memory or just the filtered group?
+    //              Using rawGroup not graph.groupByLibrary.
+
+    controller.heapGraph.rawGroupByLibrary.forEach(
+      (libraryGroup, value) {
+        final classes = value;
+        for (final theClass in classes) {
+          final shallowSize = theClass.instancesTotalShallowSizes;
+          var className = theClass.name;
+          if (shallowSize == 0 ||
+              libraryGroup == null ||
+              className == null ||
+              className == '::') {
+            continue;
+          }
+
+          // Ensure the empty library name is our group name e.g., '' -> 'src'.
+          String libraryName = theClass.libraryUri.toString();
+          if (libraryName.isEmpty) {
+            libraryName = libraryGroup;
+          }
+
+          // Map class names to familar user names.
+          final predefined =
+              predefinedClasses[LibraryClass(libraryName, className)];
+          if (predefined != null) {
+            className = predefined.prettyName;
+          }
+
+          final symbol = Symbol(
+            name: 'new $className',
+            size: shallowSize,
+            libraryUri: libraryName,
+            className: className,
+          );
+
+          Map<String, TreemapNode> currentChildren = rootChildren;
+          final TreemapNode parentReset = currentParent;
+          for (String pathPart in symbol.parts) {
+            currentChildren.putIfAbsent(
+              pathPart,
+              () {
+                final TreemapNode node = TreemapNode(
+                    name: pathPart, childrenMap: <String, TreemapNode>{});
+                currentParent.addChild(node);
+                return node;
+              },
+            );
+            currentChildren[pathPart].byteSize += symbol.size;
+            currentParent = currentChildren[pathPart];
+            currentChildren = currentChildren[pathPart].childrenMap;
+          }
+          currentParent = parentReset;
+        }
+      },
+    );
+
+    root.byteSize = root.childrenMap.values
+        .fold(0, (int current, TreemapNode node) => current + node.byteSize);
+
+    final snapshotGraph = controller.snapshots.last.snapshotGraph;
+    // Add the external heap to the heat map.
+    root.childrenMap.putIfAbsent('External Heap', () {
+      final TreemapNode node = TreemapNode(
+        name: 'External Heap',
+        childrenMap: <String, TreemapNode>{},
+      );
+      root.addChild(node);
+      node.byteSize = snapshotGraph.externalSize;
```

Comment:
instead of this add `..byteSize = snapshotGraph.externalSize;` at the end of the TreeMapNode constructor call above